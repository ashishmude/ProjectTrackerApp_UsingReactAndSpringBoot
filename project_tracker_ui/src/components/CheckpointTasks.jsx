import React from "react";
import { DataGrid } from "@material-ui/data-grid";

const columns = [
  { field: "taskName", headerName: "Task", flex: 1 },
  { field: "expectedCompletionDate", headerName: "ETA", flex: 1 },
  { field: "comments", headerName: "Comments", flex: 1 },
  {
    field: "completed",
    headerName: "Status",
    width: 160,
    valueFormatter: (params) => (params.value ? "Completed" : "In Progress"),
  },
];

const CheckpointTasks = ({ taskList }) => {
  return (
    <div style={{ height: 300, width: "100%" }}>
      <DataGrid
        rows={taskList}
        checkboxSelection={false}
        columns={columns}
        pageSize={5}
        hideFooter
      />
    </div>
  );
};
export default CheckpointTasks;
