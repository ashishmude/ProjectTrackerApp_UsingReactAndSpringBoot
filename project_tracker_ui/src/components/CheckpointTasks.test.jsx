import React from 'react';
import { shallow } from 'enzyme';
import CheckpointTasks from './CheckpointTasks';


describe("CheckpointTasks", () => {

  const data = [
      {
        id: 1,
        taskName: "Meetings with stakeholder",
        expectedCompletionDate: "25-01-2021",
        checkpointId: 1,
        comments: "Finished",
        completed: true,
      },
    ];
  it("should match the snapshot for CheckpointTasks", () => {
    const tree = shallow(<CheckpointTasks taskList={data} />);
    expect(tree.getElement()).toMatchSnapshot();
  });

});
