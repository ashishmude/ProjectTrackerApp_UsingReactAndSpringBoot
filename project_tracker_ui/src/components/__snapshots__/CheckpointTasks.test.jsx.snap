// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckpointTasks should match the snapshot for CheckpointTasks 1`] = `
<div
  style={
    Object {
      "height": 300,
      "width": "100%",
    }
  }
>
  <Memo
    checkboxSelection={false}
    columns={
      Array [
        Object {
          "field": "taskName",
          "flex": 1,
          "headerName": "Task",
        },
        Object {
          "field": "expectedCompletionDate",
          "flex": 1,
          "headerName": "ETA",
        },
        Object {
          "field": "comments",
          "flex": 1,
          "headerName": "Comments",
        },
        Object {
          "field": "completed",
          "headerName": "Status",
          "valueFormatter": [Function],
          "width": 160,
        },
      ]
    }
    hideFooter={true}
    pageSize={5}
    rows={
      Array [
        Object {
          "checkpointId": 1,
          "comments": "Finished",
          "completed": true,
          "expectedCompletionDate": "25-01-2021",
          "id": 1,
          "taskName": "Meetings with stakeholder",
        },
      ]
    }
  />
</div>
`;
